ambry-router/src/main/java/com.github.ambry.router/DecryptJob.java

```diff
@@ -51,17 +56,25 @@
    * Steps to be performed on decryption
    * 1. Fetch ContainerKey from kms for the given blob
    * 2. Decrypt encryptedKey using containerKey to obtain perBlobKey
-   * 3. Decrypt encryptedContent using perBlobKey
-   * 4. Invoke callback with the decryptedContent
+   * 3. Decrypt encryptedContent using perBlobKey if not null
+   * 4. Decrypt encryptedUserMeta using perBlobKey if not null
+   * 5. Invoke callback with the decryptedBlobContent
    */
   public void run() {
     try {
       Object containerKey = kms.getKey(blobId.getAccountId(), blobId.getContainerId());
       Object perBlobKey = cryptoService.decryptKey(encryptedKey, containerKey);
-      ByteBuffer decryptedContent = cryptoService.decrypt(enryptedContent, perBlobKey);
-      callback.onCompletion(new DecryptJobResult(blobId, decryptedContent), null);
+      ByteBuffer decryptedBlobContent = null;
+      if (encryptedBlobContent != null) {
+        decryptedBlobContent = cryptoService.decrypt(encryptedBlobContent, perBlobKey);
+      }
+      ByteBuffer decryptedUserMetadata = null;
+      if (encryptedUserMetadata != null) {
+        decryptedUserMetadata = cryptoService.decrypt(encryptedUserMetadata, perBlobKey);
+      }
+      callback.onCompletion(new DecryptJobResult(blobId, decryptedBlobContent, decryptedUserMetadata), null);
     } catch (Exception e) {
-      callback.onCompletion(new DecryptJobResult(blobId, null), e);
+      callback.onCompletion(new DecryptJobResult(blobId, null, null), e);
```

Comment:
Usually the callbacks are either-or; they either have a result or an exception. Are we creating a `DecryptJobResult` for the blob ID verification in `GetChunk`?

Reply:
yes. you are right. Will check w/ priyesh once again if this case is possible. if not, will remove the check only